tests/kernel/mem_protect/userspace/testcase.yaml

```diff
@@ -2,7 +2,15 @@ tests:
   kernel.memory_protection.userspace:
     filter: CONFIG_ARCH_HAS_USERSPACE
     tags: kernel security userspace ignore_faults
-  kernel.memory_protection.userspace.gap_filling:
+  kernel.memory_protection.userspace.gap_filling.arc:
     filter: CONFIG_ARCH_HAS_USERSPACE and CONFIG_MPU_REQUIRES_NON_OVERLAPPING_REGIONS
+    arch_whitelist: arc
+    extra_args: CONFIG_MPU_GAP_FILLING=y
+    tags: kernel security userspace ignore_faults
+  kernel.memory_protection.userspace.gap_filling.arm:
+    filter: CONFIG_ARCH_HAS_USERSPACE and CONFIG_MPU_REQUIRES_NON_OVERLAPPING_REGIONS
+    arch_whitelist: arm
+    platform_whitelist: efr32_radio_brd4180a mps2_an521 nrf5340pdk_nrf5340_cpuapp
```

Comment:
That's correct @andrewboie. In this particular case, in theory, we could do :
```
config MPU_GAP_FILLING
    bool
    depends on MPU_REQUIRES_NON_OVERLAPPING_REGIONS 
    depends on DT_MACRO_FOR_NUM_MPU_REGIONS___ > 8
```
but this would disable the GAP FILLING feature for all platforms with 8 MPU regions, and I do not want that.

What I want is t be able to filter in or filter out tests from sanity check based on DT property values. I'll file an issue.

Appreciate if we can get this patch in, which makes the mem_protect/mem_protect suite test passing on nrf5340dk